app/src/main/java/ch/epfl/sdp/blindly/user/User.kt

```diff
@@ -293,6 +310,98 @@ class User private constructor(
                 LocalDate.now()
             ).years
         }
+
+        /**
+         * Update the field of a User
+         *
+         * @param T either a String, an Int or a List<*>
+         * @param user the user to update
+         * @param field the field to update
+         * @param newValue the new value
+         * @return the updated user
+         */
+        fun <T> updateUser(user: User, field: String, newValue: T): User {
+            when (field) {
+                USERNAME -> {
+                    if (newValue !is String)
+                        throw java.lang.IllegalArgumentException("Expected newValue to be a String")
+                    user.username = newValue
+                }
+                LOCATION -> {
+                    if (newValue !is List<*>)
```

Comment:
Is this test on List<*> really checking the type ? Or does it check only if it is an object of type List ? I honestly don't know how it works. If it doesn't check the type, I think that another way of doing this would be checking the type of the first element, even if it is not perfect.
This check is done multiple times in this file, I won't comment other ones.